x-pack/plugins/ml/server/routes/system.ts

```diff
@@ -119,28 +116,34 @@ export function systemRoutes({
         }),
       },
     },
-    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
-      try {
-        const ignoreSpaces = request.query && request.query.ignoreSpaces === 'true';
-        // if spaces is disabled force isMlEnabledInSpace to be true
-        const { isMlEnabledInSpace } =
-          spacesPlugin !== undefined
-            ? spacesUtilsProvider(spacesPlugin, (request as unknown) as Request)
-            : { isMlEnabledInSpace: async () => true };
+    licensePreRoutingFactory(
+      () => {
+        cachedLicenseCheckResult = getLicenseCheckResults();
+        return cachedLicenseCheckResult;
+      },
+      async (context, request, response) => {
+        try {
+          const ignoreSpaces = request.query && request.query.ignoreSpaces === 'true';
+          // if spaces is disabled force isMlEnabledInSpace to be true
+          const { isMlEnabledInSpace } =
+            spacesPlugin !== undefined
+              ? spacesUtilsProvider(spacesPlugin, (request as unknown) as Request)
+              : { isMlEnabledInSpace: async () => true };
 
-        const { getPrivileges } = privilegesProvider(
-          context.ml!.mlClient.callAsCurrentUser,
-          xpackMainPlugin,
-          isMlEnabledInSpace,
-          ignoreSpaces
-        );
-        return response.ok({
-          body: await getPrivileges(),
-        });
-      } catch (error) {
-        return response.customError(wrapError(error));
+          const { getPrivileges } = privilegesProvider(
+            context.ml!.mlClient.callAsCurrentUser,
+            cachedLicenseCheckResult,
```

Comment:
I think it's ok to call `getLicenseCheckResults` again in here rather than caching the license object locally.
`getLicenseCheckResults` just returns and has no overhead.
`job_service.ts` is doing this without a cache.

Reply:
Updated in 609be5c4c218d1feed4b249158c438ce7c30f99d